connector-runtime/connector-runtime-core/src/main/java/io/camunda/connector/runtime/core/error/ConnectorError.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
+ * under one or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information regarding copyright
+ * ownership. Camunda licenses this file to you under the Apache License,
+ * Version 2.0; you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.camunda.connector.runtime.core.error;
+
+import static io.camunda.connector.feel.FeelConnectorFunctionProvider.*;
+
+import com.fasterxml.jackson.annotation.JsonSubTypes;
+import com.fasterxml.jackson.annotation.JsonSubTypes.Type;
+import com.fasterxml.jackson.annotation.JsonTypeInfo;
+import com.fasterxml.jackson.annotation.JsonTypeInfo.Id;
+
+@JsonTypeInfo(use = Id.NAME, property = ERROR_TYPE_PROPERTY)
+@JsonSubTypes({
+  @Type(value = BpmnError.class, name = BPMN_ERROR_TYPE_VALUE),
+  @Type(value = FailJob.class, name = FAIL_JOB_TYPE_VALUE)
```

Comment:
Thinking about this: `JobError` would maybe a better matching name as `BpmnError` is a type of error and `FailJob` is more like a command.